runtime/src/test/java/com/flipkart/flux/resource/StateMachineResourceTest.java

```diff
@@ -284,6 +284,145 @@ public void testPostScheduledEvent_withTriggerTimeInMilliSeconds() throws Except
+            assertThat(eventPostResponse1.getStatus()).isEqualTo(Response.Status.ACCEPTED.getStatusCode());
+            /* Give some time to task to execute, task thread sleeps for 1000 ms with 1 retry. */
+            Thread.sleep(4000);
+
+            //status of state should be sidelined and should be able to update event data now
+            String smId = smCreationResponse.getBody();
+            State state4 = stateMachinesDAO.findById(smId).getStates().stream().filter(e -> e.getName().equals("test_state4"))
+                    .findFirst().orElse(null);
+            assertThat(state4).isNotNull();
+            assertThat(state4.getStatus()).isEqualTo(Status.sidelined);
+            /* Make 'shouldFail' flag to false before update Event. Update Event will unsideline the dependent task
+            * after event data update. */
+            TestWorkflow.shouldFail = false;
+            String eventJson2 = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream(
+                    "updated_event_data.json"));
+            final HttpResponse<String> eventPostResponse2 = Unirest.post(
+                    STATE_MACHINE_RESOURCE_URL + SLASH + smCreationResponse.getBody() + "/context/eventupdate").header(
+                    "Content-Type", "application/json").body(eventJson2).asString();
+            assertThat(eventPostResponse2.getStatus()).isEqualTo(Response.Status.ACCEPTED.getStatusCode());
+            /* Give some time to task to execute, task thread sleeps for 1000 ms */
+            Thread.sleep(2000);
+
+            /* Assert for task completed and updated event data.*/
+            state4 = stateMachinesDAO.findById(smId).getStates().stream().filter(e -> e.getName().equals("test_state4"))
+                    .findFirst().orElse(null);
+            assertThat(state4).isNotNull();
+            assertThat(state4.getStatus()).isEqualTo(Status.completed);
+            Event event0 = eventsDAO.findBySMIdAndName(smId, "event0");
+            assertThat(event0.getEventData()).isEqualTo("50");
+        } finally {
+            TestWorkflow.shouldFail = false;
+        }
+    }
+
+    private void testEventUpdate_IneligibleTaskStatus_Util(HttpResponse<String> smCreationResponse) throws Exception {
+        String eventJson1 = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream("event_data.json"));
+        String eventJson2 = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream(
+                "updated_event_data.json"));
+
+        final HttpResponse<String> eventPostResponse1 = Unirest.post(
+                STATE_MACHINE_RESOURCE_URL + SLASH + smCreationResponse.getBody() + "/context/events").header(
+                "Content-Type", "application/json").body(eventJson1).asString();
+            /* Try update Event while task is in running state */
+        final HttpResponse<String> eventPostResponse2 = Unirest.post(
```

Comment:
Is this comment still valid, when this helper method called by multiple tests.

Reply:
Removed.